src/utils/HTTPTransport.ts

```diff
@@ -0,0 +1,79 @@
+enum Methods {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
+
+type TRequestData = Record<string, string | number>;
+
+export type TRequestOptions = {
+  method?: Methods
+  headers?: Record<string, string>
+  timeout?: number
+  data?: unknown
+};
+
+function queryStringify(data: TRequestData) {
+  if (typeof data !== 'object') {
+    throw new Error('Data must be object');
+  }
+
+  const keys = Object.keys(data);
+  return keys.reduce((result, key, index) => `${result}${key}=${data[key]}${index < keys.length - 1 ? '&' : ''}`, '?');
+}
```

Comment:
Этот код выглядит довольно сложным. Я предложу более простое решение
```suggestion
  const entries = ;
  return "?" + Object
    .entries(data)
    .map(([key, value]) => `${key}=${value}`)
    .join("&")    
}
```